Source SalesForce: move to next releaseStage (#28781)
airbytehq/airbyte: docs/integrations/sources/salesforce.md

> 

Reviewer: can you add a section that explains the differences between REST API and BULK API and the limitations when syncing from BULK API? some of the important points to highlight: 

- What differences between BULK & REST? 
- When do we use BULK or REST? 
- What are the limitations of the BULK API? e.g; which fields cannot be synced via REST?

Reviewer: Also the context you mentioned in the sheet should be added: 

```
1. docs for the api quota updated 
2. """make it clear which types of streams are not queryable and why""" We get the info about streams from salesforce DESCRIBE resource (some internal logic is used there to make it queryable or not), see https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_describeGlobal.htm
 2.1. We also exclude some of the streams which are marked as "queryable" but the error is raised when we try to get them (list was obtained experimentally).

```

Author: docs updated

The Salesforce connector is restricted by Salesforce’s [Daily Rate Limits](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm). The connector syncs data until it hits the daily rate limit, then ends the sync early with success status, and starts the next sync from where it left off. Note that picking up from where it ends will work only for incremental sync, which is why we recommend using the [Incremental Sync - Deduped History](https://docs.airbyte.com/understanding-airbyte/connections/incremental-deduped-history) sync mode.

:::tip

> Airbyte recommends to use at least [Enterpise edition](https://help.salesforce.com/s/articleView?id=sf.users_add_products_subscription_management.htm&type=5) for daily base syncs.

Reviewer: can you explain to the user why we recommend using the enterprise edition?

> You may also turn on feature `Force to use BULK API` for faster syncs.

Reviewer: Unless we have done definitive benchmarks, I don't think we should make this claim. I imagine BULK is slower for low volumes. 

```suggestion
```

The Salesforce connector is restricted by Salesforce’s [Daily Rate Limits](https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_api.htm). The connector syncs data until it hits the daily rate limit, then ends the sync early with success status, and starts the next sync from where it left off. Note that picking up from where it ends will work only for incremental sync, which is why we recommend using the [Incremental Sync - Append + Deduped](https://docs.airbyte.com/understanding-airbyte/connections/incremental-append-deduped) sync mode.

:::tip

> Airbyte recommends to use at least [Enterprise edition](https://help.salesforce.com/s/articleView?id=sf.users_add_products_subscription_management.htm&type=5) for daily base syncs. This subscription level provides the minimum required API limits.

Reviewer: ```suggestion
To use this connector, you'll need at least the Enterprise edition of Salesforce or the Professional Edition with API access purchased as an add-on. Reference the [Salesforce docs about API access](https://help.salesforce.com/s/articleView?id=000385436&type=1) for more information.
```

:::

> ### BULK API vs REST API

Reviewer: ```suggestion
### A note on the BULK API vs REST API
```

Airbyte fetches and handles all the possible and available streams dynamically based on:

- If the authenticated Salesforce user has the Role and Permissions to read and fetch objects
- If the stream has the queryable property set to true. Airbyte can fetch only queryable streams via the API. If you don’t see your object available via Airbyte, and it is queryable, check if it is API-accessible to the Salesforce user you authenticated with.

### A note on the BULK API vs REST API and their limitations

Salesforce allows extracting data using either the [BULK API](https://developer.salesforce.com/docs/atlas.en-us.236.0.api_asynch.meta/api_asynch/asynch_api_intro.htm) or [REST API](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/intro_what_is_rest_api.htm). To achieve fast performance, Salesforce recommends using the BULK API for extracting larger amounts of data (more than 2,000 records). For this reason, the Salesforce connector uses the BULK API by default to extract any Salesforce objects, unless any of the following conditions are met:

> - The Salesforce object has columns which are unsupported by the BULK API, like properties in schema: `base64` or `object`-like

Reviewer: ```suggestion
- The Salesforce object has columns which are unsupported by the BULK API, like columns with a `base64` or `complexvalue` type
```

| Version | Date       | Pull Request                                             | Subject                                                                                                                              |
|:--------|:-----------|:---------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------|

> | 2.1.2   | 2023-07-06 | [28781](https://github.com/airbytehq/airbyte/pull/28781) | Fix pagination for BULK API jobs; Addoption to force use BULK API                                                                    |

Reviewer 2: nit: only if we're keeping the new option, the changelog is missing a space between `Addoption`
